### PR TITLE
Edit and disable fields

### PR DESCRIPTION
**Pull request by vinay-ThoughtWorks**

### CODE REVIEW

File: ui/react-components/components/AppointmentEditorFooter/AppointmentEditorFooter.jsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ import UpdateButtons from "../EditAppointment/UpdateButtons.jsx";
 
 const AppointmentEditorFooter = props => {
 
-    const {checkAndSave, isEdit, showUpdateOptions} = props;
-    const[showUpdateButtons, setShowUpdateButtons] = useState(showUpdateOptions);
+    const {checkAndSave, isEdit, isOptionsRequired} = props;
```
Comment (binduak) [inline ui/react-components/components/AppointmentEditorFooter/AppointmentEditorFooter.jsx]: what does options mean here for "isOptionsRequired" props ??

File: ui/react-components/components/EditAppointment/UpdateButtons.module.scss [hunk 2]
```diff
@@ -56,7 +56,7 @@
     color: $secondary-color-1;
     background: $primary-color-2;
     &:hover {
-      background: $primary-color-2;
+      background: #454b86;
```
Comment (binduak) [inline ui/react-components/components/EditAppointment/UpdateButtons.module.scss]: should we check with UX Rohit to get the variable name for this new color

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 3]
```diff
@@ -167,6 +179,27 @@ const EditAppointment = props => {
...
+        };
+    };
+
+    const getRecurringPattern = () => {
+        const recurringPattern = {
+            type: appointmentDetails.recurrenceType,
+            period: appointmentDetails.period
+        };
+        appointmentDetails.endDateType === "After" ? recurringPattern.frequency = appointmentDetails.occurrences : recurringPattern.endDate = appointmentDetails.recurringEndDate;
+        if (appointmentDetails.recurrenceType === 'WEEK') {
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Using constants for **After, WEEK**

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 4]
```diff
@@ -199,18 +258,43 @@ const EditAppointment = props => {
...
+                endDateError: !appointmentDetails.recurringEndDate
+            })
+        } else {
+            updateErrorIndicators({
+                occurrencesError: !appointmentDetails.occurrences || appointmentDetails.occurrences < 1
+            })
+        }
+        return appointmentDetails.service && appointmentDetails.startTime && appointmentDetails.endTime && startTimeBeforeEndTime
+            && appointmentDetails.appointmentDate && isValidEndDate();
+    };
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: This function is checking for a valid recurring appointment and also updating error messages, better separating functionality.

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 5]
```diff
@@ -230,16 +314,19 @@ const EditAppointment = props => {
...
+                setOriginalOccurrences(recurringPattern.frequency);
                 updateAppointmentDetails({
                     recurrenceType: recurringPattern.type,
-                    recurringStartDate: moment(new Date(appointmentResponse.startDateTime)),
                     recurringEndDate: recurringPattern.endDate && moment(new Date(recurringPattern.endDate)),
                     occurrences: recurringPattern.frequency,
                     period: recurringPattern.period,
-                    weekDays: recurringPattern.daysOfWeek && selectWeekDays(getWeekDays(appConfig && appConfig.startOfWeek), recurringPattern.daysOfWeek)
+                    weekDays: recurringPattern.daysOfWeek && selectWeekDays(getWeekDays(appConfig && appConfig.startOfWeek), recurringPattern.daysOfWeek),
+                    endDateType: recurringPattern.endDate ? 'On' : 'After'
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Please use constants

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 6]
```diff
@@ -254,11 +341,24 @@ const EditAppointment = props => {
         defaultTime: appointmentDetails.endTime
     };
 
+    const isStartDateModified = () => !isEqual(moment(originalAppointmentDate).format('DMYYYY'), moment(appointmentDetails.appointmentDate).format('DMYYYY'));
+
+    const isEndDateModified = () => !isEqual(moment(originalRecurringEndDate).format('DMYYYY'), moment(appointmentDetails.recurringEndDate).format('DMYYYY'));
+
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Two functions do the same work, parametrize and make it one function

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 7]
```diff
@@ -254,11 +341,24 @@ const EditAppointment = props => {
...
+    const isOccurrencesModified = () => originalOccurrences !== appointmentDetails.occurrences
+
+    const isApplicableForAll = () => {
+        if (isStartDateModified())
+            return false;
+        else if (isEndDateModified() || isOccurrencesModified())
+            return false;
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: The following looks good? `return !(isStartDateModified() || isEndDateModified() || isOccurrencesModified())`

File: ui/react-components/components/EditAppointment/EditAppointment.jsx [hunk 8]
```diff
@@ -362,15 +502,30 @@ const EditAppointment = props => {
...
+                    if (isRecurringAppointment()) {
+                        if (applyForAllInd === undefined) {
+                            applyForAllInd = !isStartDateModified();
+                        }
+                        setApplyForAll(applyForAllInd);
+                        checkAndUpdateRecurringAppointments(applyForAllInd).then();
+                    } else {
+                        checkAndSave().then();
+                    }
+                }}
```
Comment (shivarach) [inline ui/react-components/components/EditAppointment/EditAppointment.jsx]: Extracting it as a method makes it more readable

File: ui/react-components/components/Service/ServiceSearch.jsx [hunk 9]
```diff
@@ -7,24 +7,32 @@ import {forEach} from 'lodash';
...
 
+    useEffect(() => {
+            setDropdownOptions(createDropdownOptions(services))
+            }, [specialityUuid]);
+
+
     const loadServices = async () => {
         const services = await getAllServices();
-        setServices(createDropdownOptions(services));
+        // setServices(createDropdownOptions(services));
```
Comment (shivarach) [inline ui/react-components/components/Service/ServiceSearch.jsx]: Overlooked?

File: ui/react-components/components/CalendarPicker/CalendarPicker.jsx [hunk 10]
```diff
@@ -14,12 +14,19 @@ import classNames from "classnames";
...
 
-    const {date, onChange, isDisabled} = props;
+    const {date, onChange, isDisabled, startDate} = props;
 
     let styles = [appointmentDatePicker];
     date ? styles.push(appointmentDatePickerSelected)
         : styles.push(appointmentDatePickerNotSelected);
-    const calendar = (<Calendar showToday={false} className={classNames(styles)}/>);
+
+    const disablePastDates = (current) => {
```
Comment (binduak) [inline ui/react-components/components/CalendarPicker/CalendarPicker.jsx]: how this "disablePastDates" different from "disablePastDates " from DatePicker.jsx file ??

File: ui/react-components/components/EditAppointment/UpdateButtons.module.scss (hunk 2 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/UpdateButtons.module.scss]: Previously on icon of removing ProviderTag also, we added new color on hover. Here also for hovering effect, we added new color code. These(hover effects) are nowhere mentioned in zeplin.

File: ui/react-components/components/AppointmentEditorFooter/AppointmentEditorFooter.jsx [hunk 11]
```diff
@@ -37,20 +37,22 @@ const AppointmentEditorFooter = props => {
...
-                        {showUpdateButtons ?  <i className={classNames("fa", "fa-times")} /> : undefined}
                     </button>
                     : <button className={classNames(button, save)} onClick={checkAndSave} data-testid="check-and-save">
                         <i className={classNames("fa", "fa-check")}/>
                         <span>
                         <FormattedMessage id={'APPOINTMENT_CREATE_CHECK_AND_SAVE'} defaultMessage={'Check and Save'}/>
                     </span>
                     </button>}
-                {showUpdateButtons ? <UpdateButtons/> : undefined}
+                {isOptionsRequired && showUpdateButtons ? <UpdateButtons updateOptionsVisibleStatus={setShowUpdateButtons} checkAndSave={applyForAll =>  checkAndSave(applyForAll)} /> : undefined}
```
[the remark this answers is not in the document]
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/AppointmentEditorFooter/AppointmentEditorFooter.jsx]: Discussed

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 3 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: done

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 5 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: DONE

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 6 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: DONE

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 7 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: DONE

File: ui/react-components/components/Service/ServiceSearch.jsx (hunk 9 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/Service/ServiceSearch.jsx]: DONE

File: ui/react-components/components/CalendarPicker/CalendarPicker.jsx (hunk 10 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/CalendarPicker/CalendarPicker.jsx]: done

File: ui/react-components/components/EditAppointment/EditAppointment.jsx (hunk 8 above)
Comment (vinay-ThoughtWorks) [reply ui/react-components/components/EditAppointment/EditAppointment.jsx]: DONE

**Pull request merged**